src/ray/gcs/gcs_actor_manager.h

```diff
@@ -515,6 +518,14 @@ class GcsActorManager : public rpc::ActorInfoGcsServiceHandler {
   // Currently protects actor_to_register_callbacks_.
   ThreadChecker thread_checker_;
 
+  /// Fallback timers for graceful actor shutdowns, keyed by WorkerID.
+  /// Canceled when actors exit or restart to prevent force-killing wrong instances.
```

Comment:
what does force-killing wrong instances mean? do you mean sending the kill message to the raylet for the wrong worker? the raylet should (and needs to) handle that rather than relying on best-effort timer cancelation